plugins/woocommerce-blocks/assets/js/atomic/utils/register-product-block-type.ts

```diff
@@ -14,34 +14,53 @@ import { subscribe, select } from '@wordpress/data';
 import { isNumber, isEmpty } from '@woocommerce/types';
 
 /**
- * Configuration object for registering a block.
+ * Settings for product block registration.
  *
- * @typedef {Object} BlockConfig
- * @property {string}                               blockName               - The name of the block to register
- * @property {(string|Partial<BlockConfiguration>)} [blockMetadata]         - Block metadata or name
- * @property {Partial<BlockConfiguration>}          blockSettings           - Block settings configuration
- * @property {boolean}                              [isVariationBlock]      - Whether this block is a variation
- * @property {string}                               [variationName]         - The name of the variation if applicable
- * @property {boolean}                              isAvailableOnPostEditor - Whether the block should be available in post editor
+ * @typedef {Object} ProductBlockSettings
+ * @property {boolean} [isVariationBlock]      - Whether this block is a variation
+ * @property {string}  [variationName]         - The name of the variation if applicable
+ * @property {boolean} isAvailableOnPostEditor - Whether the block should be available in post editor
  */
-type BlockConfig = {
-	blockName: string;
-	blockMetadata?: string | Partial< BlockConfiguration >;
-	blockSettings: Partial< BlockConfiguration >;
+type ProductBlockSettings = {
 	isVariationBlock?: boolean;
-	variationName?: string;
+	variationName?: string | undefined;
 	isAvailableOnPostEditor: boolean;
 };
 
 /**
- * Manages block registration and unregistration for WordPress blocks in different contexts.
+ * Internal block config type used by the BlockRegistrationManager
+ *
+ * @typedef {Object} ProductBlockConfig
+ * @property {string}                      blockName            - The name of the block
+ * @property {Partial<BlockConfiguration>} settings             - Block settings configuration
+ * @property {ProductBlockSettings}        productBlockSettings - Product block settings
+ */
+type ProductBlockConfig = ProductBlockSettings & {
+	blockName: string;
+	settings: Partial< BlockConfiguration >;
```

Comment:
I am not convinced that this should be partial given that `ProductBlockSettings` has already optional keys

Reply:
If we want to keep the `isAvailableOnPostEditor` optional, let's update the `ProductBlockSettings`:

```ts
type ProductBlockSettings = {
	isVariationBlock?: boolean;
	variationName?: string | undefined;
	isAvailableOnPostEditor?: boolean;
};
```

Reply:
>I am not convinced that this should be partial given that ProductBlockSettings has already optional keys

@gigitux can you explain please? `BlockConfiguration` is Gutenbergs block config.

Reply:
Ok, I missed that. All good so :D